bindings/java/src/org/sleuthkit/datamodel/DataArtifact.java

```diff
@@ -49,13 +49,14 @@
 	 * @param reviewStatus     The review status of this artifact.
 	 * @param osAccount        OsAccount associated with this artifact, may be
 	 *                         null.
+	 * @param isNew            The object is newly created.
 	 */
-	DataArtifact(SleuthkitCase sleuthkitCase, long artifactID, long sourceObjId, long artifactObjId, long dataSourceObjId, int artifactTypeID, String artifactTypeName, String displayName, ReviewStatus reviewStatus, OsAccount osAccount) {
-		super(sleuthkitCase, artifactID, sourceObjId, artifactObjId, dataSourceObjId, artifactTypeID, artifactTypeName, displayName, reviewStatus);
+	DataArtifact(SleuthkitCase sleuthkitCase, long artifactID, long sourceObjId, long artifactObjId, long dataSourceObjId, int artifactTypeID, String artifactTypeName, String displayName, ReviewStatus reviewStatus, OsAccount osAccount, boolean isNew) {
+		super(sleuthkitCase, artifactID, sourceObjId, artifactObjId, dataSourceObjId, artifactTypeID, artifactTypeName, displayName, reviewStatus, isNew);
```

Comment:
@bcarrier, @jayaramcs and @raman-bt...

What is the use case for the  isNew flag?

Reply:
The isNew flag is not new (no pun intended)   I'd just missed it when I first created the DataArtifact class. 

I think the intent of the flag is to indicate to client that "this artifact was just created and hence has all its attributed loaded in and don't need to go fetch them from the database"